File: app/src/main/java/modularity/andres/it/coderdojo/ui/list/dagger/EventListModule.java
package modularity.andres.it.coderdojo.ui.list.dagger;

import dagger.Binds;
import dagger.Module;
import modularity.andres.it.coderdojo.MainActivity;
import modularity.andres.it.coderdojo.ui.list.mvp.DojoEventsListView;

@Module
public abstract class EventListModule {

    @Binds
    abstract DojoEventsListView provideFeatureView(MainActivity mainActivity);

}

File: app/src/main/java/modularity/andres/it/coderdojo/gui/list/dagger/EventListModule.kt
package modularity.andres.it.coderdojo.gui.list.dagger

import android.app.Activity
import dagger.Binds
import dagger.Module
import dagger.android.ActivityKey
import dagger.android.AndroidInjector
import dagger.multibindings.IntoMap
import modularity.andres.it.coderdojo.gui.list.EventListActivity


/**
 * Created by garu on 15/11/17.
 */

@Module(subcomponents = arrayOf(EventListActivitySubcomponent::class))
abstract class EventListModule {

    @Binds
    @IntoMap
    @ActivityKey(EventListActivity::class)
    internal abstract fun eventListInjectorFactory(builder: EventListActivitySubcomponent.Builder): AndroidInjector.Factory<out Activity>

}